lib/mesh.js

```diff
@@ -975,6 +980,63 @@ Mesh.prototype._initializeRootWebRoutes = function(callback) {
   callback();
 };
 
+Mesh.prototype.__getMiddlewareHandlers = function(middleware) {
+  const _this = this;
+  if (!middleware) throw new Error('undefined middleware in web config');
+  if (typeof middleware === 'function') return [middleware];
+  if (middleware.component == null) throw new Error(`web.middleware.component null in web config`);
+  if (middleware.methods == null || !middleware.methods.length)
+    throw new Error(
+      `web.middleware.methods empty or null in web config, component: ${middleware.component}`
+    );
+
+  const methods = middleware.methods.map(methodName => {
+    return function(req, res, next) {
+      if (!_this.exchange[middleware.component]) {
+        return next(new Error(`Component ${middleware.component} not found for global middleware`));
+      }
+      if (!_this.exchange[middleware.component][methodName]) {
+        return next(
+          new Error(`Method ${middleware.component}.${methodName} not found for global middleware`)
+        );
+      }
+      _this.exchange[middleware.component][methodName].call(
+        _this.exchange[middleware.component],
+        req,
+        res,
+        next
+      );
+    };
+  });
+  //error handler for failed middleware
+  //eslint-disable-next-line
+  methods.push(function(err, req, res, next) {
```

Comment:
question on connect - does it treat a middleware differently if it has the (err, req, res, next) signature?

Reply:
i.e. wondering what stops the error handler from being invoked under normal circumstances

Reply:
Based on its arguments,it is not invoked, see: https://github.com/senchalabs/connect#error-middleware